runtime/compiler/trj9/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -5575,8 +5575,8 @@ static void genHeapAlloc(TR::Node *node, TR::Instruction *&iCursor, TR_OpaqueCla
                uintptrj_t mask;
                if (TR::Compiler->target.is64Bit())
                   {
-                  mask = CONSTANT64(0xFFFFFFFFFFFFFFFF) << (27 - leadingZeroes(maxSafeSize));
-                  iCursor = generateTrg1Src1Imm2Instruction(cg, TR::InstOpCode::rldicr_r, node, temp2Reg, enumReg, condReg, 0, mask, iCursor);
+                  mask = 0xFFFFFFFF << (27 - leadingZeroes(maxSafeSize));
+                  iCursor = generateTrg1Src1Imm2Instruction(cg, TR::InstOpCode::rlwinm_r, node, temp2Reg, enumReg, condReg, 0, mask, iCursor);
```

Comment:
embedding the comment:
Minor: rlwinm essentially consumes a 32bit mask. you have a 64-bit mask there in 64bit mode. fortunately for rlwinm, binary encoding takes only the lower portion. it just works out.

Reply:
All method definitions define the mask as a 64bit value:
```
omr/compiler/p/codegen/GenerateInstructions.hpp:

234 TR::Instruction *generateTrg1Src1Imm2Instruction(
  1                    TR::CodeGenerator      *cg,
  2                    TR::InstOpCode::Mnemonic   op,
  3                    TR::Node        *n,
  4                    TR::Register    *treg,
  5                    TR::Register    *s1reg,
  6                    int32_t         imm1,
  7                    int64_t         imm2,
  8                    TR::Instruction *preced = 0);
  9
 10 TR::Instruction *generateTrg1Src1Imm2Instruction(
 11                    TR::CodeGenerator      *cg,
 12                    TR::InstOpCode::Mnemonic   op,
 13                    TR::Node        *n,
 14                    TR::Register    *treg,
 15                    TR::Register    *s1reg,
 16                    TR::Register    *cr0reg,
 17                    int32_t         imm1,
 18                    int64_t         imm2,
 19                    TR::Instruction *preced = 0);
```

There are also other cases in the code base where rlwinm is used with a 64bit mask. How should I proceed?

Reply:
it is acceptable as it is, as i indicated above.

Reply:
initially i had a concern about what really ended up in "mask" variable by the expression: (0xFFFFFFFF << shift-amount).